objects/src/accounts/storage/mod.rs

```diff
@@ -376,20 +379,8 @@ mod tests {
         // storage with values for default types
         let storage = AccountStorage::new(
             vec![
-                SlotItem {
-                    index: 0,
-                    slot: StorageSlot {
-                        slot_type: StorageSlotType::default(),
-                        value: [ONE, ONE, ONE, ONE],
-                    },
-                },
-                SlotItem {
-                    index: 2,
-                    slot: StorageSlot {
-                        slot_type: StorageSlotType::default(),
-                        value: [ONE, ONE, ONE, ZERO],
-                    },
-                },
+                SlotItem::new_value(0, 0, [ONE, ONE, ONE, ONE]),
+                SlotItem::new_value(2, 0, [ONE, ONE, ONE, ZERO]),
```

Comment:
I'm not super found of `0, 0`, since it is a bit obscure what these values mean without IDE support. But adding wrappers types on this PR would be a lot more changes, so for now this is mostly to make the code more compact, and to make the arity explicit.

Reply:
I think the need for `index` will go away as a part of https://github.com/0xPolygonMiden/miden-base/issues/667 refactoring. Specifically, if we represent storage commitment not as a Merkle tree of depth 8 but as a sequential hash of storage slots, the need for `SlotItem` struct will go away and `AccountStorage` may end up looking like this:

```Rust
pub struct AccountStorage {
    slots: Vec<StorageSlot>,
    commitment: Digest,
}

pub enum StorageSlot {
    /// Value of arity 0
    SimpleValue(Word),
    /// Value of arity > 0
    ComplexValue(Vec<Word>),
    Array(StorageArray),
    Map(StorageMap),
}
```

Reply:
cc @phklive